File: 147/solution.cpp
#include<iostream>
int cal(int a, int b) {
	if(a>b) std::swap(a, b);
	a<<=1, b<<=1;

	int ret = 0;
	// parallel
	for(int i = 0;i<a;i+=2) {
		for(int j = 0;j<b;j+=2) {
			for(int l = i+1;l<=a;l+=2) {
				ret += (b-j)/2;
			}
		}
	}

	// vertical
	for(int i = 0;i<a;i++) {
		for(int j = i&1;j<b;j+=2) {
			for(int l = std::min(j, a-i);l>=1;l--) {
				int upper = std::min(a-i-l+1, b-j+1);
				ret+=upper-std::max(1, l-j);
			}
		}
	}
	
	return ret;
}
int main() {
	int ans = 0;
	for(int i = 1;i<=47;i++) {
		for(int j = 1;j<=43;j++) {
			ans += cal(i, j);
		}
	}
	std::cout<<ans<<std::endl;
	return 0;
}

File: 147/solution.py
def cal(a, b):
	if a > b:
		a, b = b, a
	a<<=1
	b<<=1

	ret = 0
	for i in range(0, a, 2):
		for j in range(0, b, 2):
			for l in range(i+1, a+1, 2):
				ret += (b-j)//2
	for i in range(a):
		for j in range(i&1, b, 2):
			for l in range(min(j, a-i), 0, -1):
				ret += min(a-i-l+1, b-j+1) - max(1, l-j)
	return ret

if __name__ == '__main__':
	ans = 0
	for i in range (1, 48):
		for j in range(1, 44):
			ans += cal(i, j)
	print(ans)
